appservice/src/SiteWrapper.ts

```diff
@@ -584,14 +590,18 @@ export class SiteWrapper {
             isMercurial: false
         };
 
-        this.log(outputChannel, `"${this.appName}" is being connected to GitHub repo. This may take several minutes...`);
+        this.log(outputChannel, `"${this.appName}" is being connected to the GitHub repo. This may take several minutes...`);
         try {
-            await client.webApps.createOrUpdateSourceControlWithHttpOperationResponse(this.resourceGroup, this.name, siteSourceControl);
+            await this.updateSourceControl(client, siteSourceControl);
         } catch (err) {
             try {
                 // a resync will fix the first broken build
                 // https://github.com/projectkudu/kudu/issues/2277
-                await client.webApps.syncRepository(this.resourceGroup, this.name);
+                if (this.slotName) {
```

Comment:
nit: How about you move this into a SiteWrapper.syncRepository method

Reply:
I'm not sure where we'd use this anywhere else (which is why I didn't create its own method).